File: uniform_product_distribution.py
import random

def uniform_product_random_number():
  return int(random.random() * random.random() * 10)

scores = {0: 0, 1: 0, 2: 0, 3: 0, 4: 0, 5: 0, 6: 0, 7: 0, 8: 0, 9: 0}
for i in range(0, 100000):
  scores[uniform_product_random_number()] += 1

for index, score in scores.iteritems():
  print index, '*' * int(score / 100000.0 * 100)

File: uniform_product_distribution.rb
def uniform_product_random_number
  (Random.rand * Random.rand * 10).floor
end

scores = Hash[(0...10).zip(Array.new(10, 0))]
100_000.times do
  scores[uniform_product_random_number] += 1
end

scores.each do |index, hits|
  puts "#{index}: #{'*' * (hits / 100_000.0 * 100).to_i}"
end
